README.md

```diff
@@ -8,6 +8,36 @@ This project holds various deployment scripts that are used with AWS CodeDeploy
 
 This will check if the `package.json` version has been updated for a project and then create a new release on github after running `npm build` and commiting any changed release assets in the `dist` directory.
 
+### `cachedInstallModules.sh`
+
+This can speed up installation of node modules on hosted CI. It stores a hash of the `package.json` into the node_modules directory and uses it to determine if we may need to reinstall. Alongside this, it can automatically upgrade `npm`, and wipe out the cached node_modules directory if `package.json` has changed (to ensure unused modules are removed, and that new releases satisfying the current version range are updated to).
+
+Use and configure with package.json:
+
+```
+  "scripts": {
+    "install:cached": "rm -rf ./deployment-helpers && git clone https://github.com/holidayextras/deployment-helpers.git && ./deployment-helpers/nodeApps/cachedInstallModules.sh"
+  },
+  "config": {
+    "nodeModuleCaching": {
+      // "clear", "prune" or omit.
+      // "clear" is slower, but will pick up all available updates to modules
+      // "prune" is faster, but will continue using already-cached modules if
+      //   they still satisfy the range. Extraneous modules are removed.
+      // If omitted, the usual approach of simply running `npm install` is used,
+      // which will not install available updates or remove extraneous modules.
+      "strategy": "clear",
+
+      // Which npm version to install. Omit to use the existing default.
+      "npmVersion": 3,
```

Comment:
Possibly, it doesn't because

- The `engines` field can include more complex version ranges like `>=1 <3` which can't always simply be pushed back into `npm install`
- The `dependencies` field could work, but most projects don't specify `npm` as a local dependency (and they could, but then the install time increases as it ends up installed both locally+globally, your initial dependency tree is influenced by npm's own dependencies, and you have to manipulate `$PATH` to guarantee every command in your build uses it)